### PR TITLE
[libc][stdfix] Initial support for stdfix.h and fixed point arithmetic.

### PR DESCRIPTION
**Pull request by lntue**
This is just a draft to see that what we need to build and test functions using `_Fract` and `_Accum` types.  Will split them into smaller patches to commit.

### CODE REVIEW

**Comment by lntue** [comment]
Build and test with x86-64 clang (trunk) works with latest version.

**Comment by lntue** [comment]
Spinoff 1: First patch to generate `stdfix.h` header and add compiler support detection https://github.com/llvm/llvm-project/pull/81255

File: libc/src/__support/CPP/type_traits/is_fixed_point.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  template <typename Head, typename... Args>
+  LIBC_INLINE_VAR static constexpr bool __is_unqualified_any_of() {
+    return (... || is_same_v<remove_cv_t<Head>, Args>);
+  }
+
+public:
+  LIBC_INLINE_VAR static constexpr bool value = __is_unqualified_any_of<
+      T, short fract, fract, long fract, unsigned short fract, unsigned fract,
+      unsigned long fract, short accum, accum, long accum, unsigned short accum,
+      unsigned accum, unsigned long accum>();
```
Comment (PiJoules) [inline libc/src/__support/CPP/type_traits/is_fixed_point.h:33]: We'll also want to cover the saturation equivalents of each of these.

Comment (lntue) [reply libc/src/__support/CPP/type_traits/is_fixed_point.h:33]: Done.

**Comment by lntue** [comment]
Spinoff 4:  Add support for fixed point types in the testing infrastructure. #81819 
Spinoff 5: Add abs functions for signed fixed point types. #81823

**Pull request closed without merging**